onnxscript/rewriter/ort_fusions/sdpa_test.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""SDPA fusion test cases."""
+
+from __future__ import annotations
+
+import math
+import unittest
+
+import numpy
+
+import onnxscript.ir as ir
+import onnxscript.optimizer
+from onnxscript import script
+from onnxscript.onnx_opset import opset18 as op
+from onnxscript.onnx_types import FLOAT
+from onnxscript.rewriter.ort_fusions.sdpa import fuse_sdpa
+
+batchsize = 2
+num_heads = 4
+seq_len = 8
+headsize = 128
+scale_factor = math.sqrt(headsize)
+sqrt_scale_factor = math.sqrt(scale_factor)
+
+
+@script()
+def _masked_pre_div_sdpa_script(query, key, value, mask):
+    key_transposed = op.Transpose(key, perm=[0, 1, 3, 2])
+    divisor = op.Constant(value_float=sqrt_scale_factor)
+    scaled_query = op.Div(query, divisor)
+    scaled_key = op.Div(key_transposed, divisor)
+    attn_score = op.MatMul(scaled_query, scaled_key)
+    masked_attn_score = op.Add(attn_score, mask)
+    attn_weight = op.Softmax(masked_attn_score, axis=-1)
+    attn_output = op.MatMul(attn_weight, value)
+    return attn_output
+
+
+class _MaskedPreDivSDPATestCase:
+    def get_onnx_model(self):
+        if not hasattr(self, "_onnx_model"):
+            qkv_type = FLOAT[batchsize, num_heads, seq_len, headsize]
```

Comment:
Maybe we can add a test model for 3d inputs? I'm curious to see how the check function behaves for 4d/3d inputs

Reply:
Sure, maybe you can add it as part of your extension? First, the checks will need to be added to the rule (as you can see in the comments there, they are currently missing.)

Reply:
Sounds good will do.